Dockerfile

```diff
@@ -0,0 +1,21 @@
+# NOTE(dperny): for some reason, alpine was giving me trouble
+FROM golang:1.10.3-stretch
+
+RUN apt-get install make git gcc
+
+# This is sort of a workaround... basically, we want to be able to install all
+# of the dependencies specified in `make setup`. But if we wait until after we
+# copy code over, we'll bust the cache and reinstall these dependencies every
+# time, which would be bad. This is a common problem in docker.
+RUN go get -u github.com/golang/lint/golint &&\
+  go get -u github.com/gordonklaus/ineffassign &&\
+  go get -u github.com/client9/misspell/cmd/misspell &&\
+  go get -u github.com/lk4d4/vndr &&\
+  go get -u github.com/stevvooe/protobuild
+
+WORKDIR /go/src/github.com/docker/swarmkit/
+COPY . .
```

Comment:
if you are using local machine you dont need to copy entire source code. you can mount the folders when you run container. this will be useful only when we use circleCI2.0 as they dont support mount option now.

Reply:
right, of course, but i think it's better to copy the whole deal so it's a proper swarmkit image, instead of just a fancy go image.